pkg/gadgets/trace/open/tracer/gadget.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022-2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer
+
+import (
+	gadgetregistry "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-registry"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/open/types"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/parser"
+)
+
+type GadgetDesc struct{}
+
+func (g *GadgetDesc) Name() string {
+	return "open"
+}
+
+func (g *GadgetDesc) Category() string {
+	return gadgets.CategoryTrace
+}
+
+func (g *GadgetDesc) Type() gadgets.GadgetType {
+	return gadgets.TypeTrace
+}
+
+func (g *GadgetDesc) Description() string {
+	return "opensnoop traces open() system calls"
```

Comment:
```suggestion
	return "Trace open system calls"
```

Reply:
Just a quick notice regarding these descriptions: I copied them from the gadget-collection to be consistent (those go into the docs afaik). I see that it's not the best option in many cases (and I'll change these to the change requests / cmd/common counterparts), but we still should revisit them sometime.